source/common/umf_helpers.hpp

```diff
@@ -160,6 +161,39 @@ template <typename Type> umf_result_t &getPoolLastStatusRef() {
     return last_status;
 }
 
+/// @brief translates UMF return values to UR.
+/// This function assumes that the native error of
+/// the last failed memory provider is ur_result_t.
+ur_result_t umf2urResult(umf_result_t umfResult) {
```

Comment:
probably this function should be 'inline' - to fix the compilation error you got.